proxy-auth.ks.in

```diff
@@ -28,9 +28,25 @@ if [[ ! -f /mnt/sysroot/etc/yum.repos.d/addon.repo ]]; then
 fi
 
 # Check that the proxy configuration was written to the repo file
-grep -q 'proxy=http://anaconda:qweqwe@PROXY-ADDON' /mnt/sysroot/etc/yum.repos.d/addon.repo
-if [[ $? -ne 0 ]]; then
+# The repo configuration has a different format on RHEL-10+/Fedora 40+
+os_version=$(. /etc/os-release && echo ${VERSION_ID%%.*})
+platform=$(. /etc/os-release && echo ${ID})
+
+if [ "${platform}" == "rhel" ] && [ ${os_version} -lt 10 ]; then
```

Comment:
Regarding the per-platform conditions, there is an open PR https://github.com/rhinstaller/kickstart-tests/pull/1174 which adds get_platform() function that could be used similarly as in [repo-install](https://github.com/rvykydal/kickstart-tests/blob/d5b2933709087e29ec10d81b5ae3cfbb1730c257/repo-install.ks.in#L21) test when the PR is merged. I hope we merge it soon so you can use it. Or I can take care of it in a separate PR.

Reply:
The https://github.com/rhinstaller/kickstart-tests/pull/1174 is merged.

Reply:
Thanks for the suggestion! In the end I realized I can actually use the substituted values of OS name and version (I'd have to have them separated in the condition expressions anyway), so I used this approach instead of `get_platform()` - I hope it's OK.